### PR TITLE
chore(cicd): update buildspec to list and build jobs

### PR DESCRIPTION
**Pull request by bvtujo**
Updates the buildspec to list, package, and build jobs. Shouldn't be merged in until #1428 is completed.

Related #1131 #1428 #1480 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/cicd/buildspec.yml [hunk 1]
```diff
@@ -32,17 +34,22 @@ phases:
...
           ./copilot-linux svc package -n $svc -e $env --output-dir './infrastructure' --tag $tag;
           done;
+          for job in $jobs; do
+          ./copilot-linux job package -n $job -e $env --output-dir './infrastructure' --tag $tag;
+          done;
         done;
       - ls -lah ./infrastructure
+      # Concatenate jobs and services into one var for addons
       # If addons exists, upload addons templates to each S3 bucket and write template URL to template config files.
+      - JOBS_SVCS=$(echo $jobs $svcs)
```
Comment (bvtujo) [inline templates/cicd/buildspec.yml]:
```suggestion
      # Concatenate jobs and services into one var for addons
      - JOBS_SVCS=$(echo $jobs $svcs)
      # If addons exists, upload addons templates to each S3 bucket and write template URL to template config files.
```

File: templates/cicd/buildspec.yml [hunk 2]
```diff
@@ -32,17 +34,22 @@ phases:
           for svc in $svcs; do
           ./copilot-linux svc package -n $svc -e $env --output-dir './infrastructure' --tag $tag;
           done;
+          for job in $jobs; do
+          ./copilot-linux job package -n $job -e $env --output-dir './infrastructure' --tag $tag;
+          done;
         done;
       - ls -lah ./infrastructure
+      # Concatenate jobs and services into one var for addons
+      - JOBS_SVCS=$(echo $jobs $svcs)
```
Comment (efekarakus) [inline templates/cicd/buildspec.yml]:
```suggestion
      # Concatenate jobs and services into one var for addons
      - WORKLOADS=$(echo $jobs $svcs)
```

File: templates/cicd/buildspec.yml [hunk 3]
```diff
@@ -32,17 +34,22 @@ phases:
...
       # If addons exists, upload addons templates to each S3 bucket and write template URL to template config files.
       - |
-        for svc in $svcs; do
-          ADDONSFILE=./infrastructure/$svc.addons.stack.yml
+        for js in $JOBS_SVCS; do
```
Comment (efekarakus) [inline templates/cicd/buildspec.yml]:
```suggestion
        for workload in $WORKLOADS; do
```

**Review by efekarakus — COMMENTED** [review]
Looks good to me! Just a suggestion on renaming `JOBS_SVCS` to `WORKLOADS` or just `workloads`

**Review by efekarakus — APPROVED** [review]

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**